src/LayerDicts.jl

```diff
@@ -66,37 +64,19 @@ end
 
 Base.length(ld::LayerDict) = length(keys(ld))
 
-if VERSION < v"0.7.0-DEV.5126"
-    function Base.start(ld::LayerDict)
-        ld_keys = keys(ld)
-        return (ld_keys, start(ld_keys))
-    end
+function Base.iterate(ld::LayerDict, state=(keys(ld),))
+    ld_keys = first(state)
+    key_state = Base.tail(state)
 
-    function Base.done(ld::LayerDict, state)
-        ld_keys, key_state = state
-        return done(ld_keys, key_state)
-    end
+    iter = iterate(ld_keys, key_state...)
 
-    function Base.next(ld::LayerDict, state)
-        ld_keys, key_state = state
-        key, new_key_state = next(ld_keys, key_state)
-        return (key => ld[key]), (ld_keys, new_key_state)
+    if iter === nothing
+        return nothing
     end
```

Comment:
I would shorten the  above `if` into  `iter === nothing && return nothing`.
But I guess that is just personal preference, so do not feel compelled to make this change if you think the `if` is cleaner.